docs/python_support/reference.md

```diff
@@ -0,0 +1,769 @@
+
+ Check out the [quickstart guide](quickstart.md) for a simple example that shows how to use this feature.
+
+ ## Building Docker Image
+
+The publicly available `openvino/model_server` image on Docker Hub supports Python, but does not come with external modules installed. If Python is all you need then you can use the public image without modification. Otherwise, you will need to extend the public image with additional layers that install any modules required for your Python code to run. For example, let's say your code requires numpy.
+In that case, your Dockerfile may look like this:
+
+```dockerfile
+FROM openvino/model_server:latest
+USER root
+ENV LD_LIBRARY_PATH=/ovms/lib
+ENV PYTHONPATH=/ovms/lib/python
+RUN apt update && apt install -y python3-pip git
+RUN pip3 install numpy
+ENTRYPOINT [ `/ovms/bin/ovms` ]
+```
+
+You can also modify `requirements.txt` from our [python demos](../../demos/python_demos) and from repository top level directory run `make python_image` 
+
+## `OvmsPythonModel` class
+
+When deploying a Python node, the Model Server expects a Python file with an `OvmsPythonModel` class implemented:
+
+```python
+class OvmsPythonModel:
+
+    def initialize(self, kwargs):
+        """
+        `initialize` is called when model server loads graph definition. 
+        It allows to initialize and maintain state between subsequent execute() calls and even graph instances. 
+        For gRPC unary, graphs are recreated per request. 
+        For gRPC streaming, there can be multiple graph instances existing at the same time. 
+        OvmsPythonModel object is initialized with this method and then shared between all graph instances. 
+        Implementing this function is optional.
+
+        Parameters:
+        -----------
+        kwargs : dict
+            Available arguments:
+            * node_name: string
+                Name of the node in the graph
+            * input_streams: list of strings
+                List of input stream names defined for the node in graph
+                configuration
+            * output_streams: list of strings
+                List of output stream names defined for the node in graph
+                configuration
+        -----------
+        """
+        print("Running initialize...")
+
+    def execute(self, inputs):
+        """
+        `execute` is called in `Process` method of PythonExecutorCalculator
+        which in turn is called by the MediaPipe framework. How MediaPipe
+        calls the `Process` method for the node depends on the configuration
+        and the two configurations supported by PythonExecutorCalculator are:
+        
+        * Regular: `execute` runs every time the node receives all inputs with
+        the same timestamp. Produces one set of outputs per one set of inputs.
+        For unary endpoints it's the only possible configuration.
+        
+        * Generative: `execute` is called once, when all inputs with the same
```

Comment:
The same for generative, here I am sure it is not always true:
Set SyncSetInputStreamHandler so that there are 3 sets: A, B ; C, D ; loopback
This way all inputs must not be provided

Another misguidance is that execute can be called only once for generative - current implementation resets the generator after first generation is done, so that subsequent request can trigger new generation, please confirm